tests/test_harness_notifications.py

```
@@ -0,0 +1,135 @@
+        config.slack_bot_token = None
+        config.slack_channel = None
+        config.github_token = None
+        result = _build_notify_config(config)
+        assert result["slack_webhook"] == "https://hooks.slack.com/test"
+
+    def test_returns_multiple_keys(self):
+        config = MagicMock()
+        config.notify_slack_webhook = "https://hooks.slack.com/test"
+        config.notify_discord_webhook = "https://discord.com/api/webhooks/test"
+        config.notify_email = None
+        config.slack_bot_token = "xoxb-token"
```

Ruff (0.14.14)

[error] Possible hardcoded password assigned to: "slack_bot_token"

(S105)